Modix.Services/Quote/MessageLinkBehavior.cs

```diff
@@ -70,33 +79,29 @@ private async Task OnMessageReceivedAsync(SocketMessage message)
             }
         }
 
-        private async Task SendQuoteEmbedAsync(ulong originalChannelId, ulong messageId, SocketGuildUser quoter, ISocketMessageChannel targetChannel)
+        private async Task SendQuoteEmbedAsync(IMessage message, SocketGuildUser quoter, ISocketMessageChannel targetChannel)
         {
-            var channel = DiscordClient.GetChannel(originalChannelId);
-
-            if (channel is IGuildChannel &&
-                channel is ISocketMessageChannel messageChannel)
+            await SelfExecuteRequest<IQuoteService>(async quoteService =>
             {
-                var msg = await messageChannel.GetMessageAsync(messageId);
-                if (msg == null) { return; }
+                var embed = quoteService.BuildQuoteEmbed(message, quoter);
+                if (embed == null) return;
 
-                await SelfExecuteRequest<IQuoteService>(async quoteService =>
-                {
-                    var embed = quoteService.BuildQuoteEmbed(msg, quoter)?.Build();
-                    if (embed == null) { return; }
+                embed.Fields.First(d => d.Name == "Quoted by")
+                .Value += $" from **[#{message.Channel.Name}]({message.GetJumpUrl()})**";
```

Comment:
Should this just be part of the `BuildQuoteEmbed()` logic?

Reply:
Yep.

Reply:
I'll change that in this PR and take a closer look at the entire `QuoteService` later on today, just a quick glance on it and saw a few improvements that could be made.